pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -346,19 +347,53 @@ func (h *NetworkUtilsHandler) GenerateRandomMac() (net.HardwareAddr, error) {
 }
 
 func (h *NetworkUtilsHandler) CreateTapDevice(tapName string, isMultiqueue bool, launcherPID string) error {
-	args := []string{"tuntap", "add", "mode", "tap", "user", "qemu", "group", "qemu", "name", tapName}
-	if isMultiqueue {
-		args = append(args, "multi_queue")
+	currentHandlerSELinuxLabel, err := getProcessCurrentSELinuxLabel(fmt.Sprintf("%d", os.Getpid()))
+	if err != nil {
+		return fmt.Errorf("error reading virt-handler selinux label: %v", err)
 	}
-	cmd := exec.Command("ip", args...)
-	err := cmd.Run()
+
+	launcherSELinuxLabel, err := getProcessCurrentSELinuxLabel(launcherPID)
 	if err != nil {
-		return fmt.Errorf("failed to create tap device %s; %v", tapName, err)
+		return fmt.Errorf("error reading virt-launcher PID %s selinux label; %v", tapName, err)
+	}
+
+	createTapDeviceArgs := []string{
+		"create-tap",
+		"--tap-name", tapName,
+		"--uid", "107",
+		"--gid", "107",
+	}
+	if isMultiqueue {
+		createTapDeviceArgs = append(createTapDeviceArgs, "--multiqueue")
+	}
+	cmd := exec.Command("virt-chroot", createTapDeviceArgs...)
+
+	runtime.LockOSThread()
+	if err = selinux.SetExecLabel(launcherSELinuxLabel); err != nil {
+		return fmt.Errorf("failed to switch selinux context to %s. Reason: %v", launcherSELinuxLabel, err)
 	}
-	log.Log.Infof("Created tap device: %s", tapName)
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to create tap device %s: %v", tapName, err)
+	}
+
+	if err = selinux.SetExecLabel(currentHandlerSELinuxLabel); err != nil {
+		return fmt.Errorf("failed to switch selinux context to %s. Reason: %v", launcherSELinuxLabel, err)
+	}
+	runtime.UnlockOSThread()
```

Comment:
I think this should be `defer`ed, thus ensuring this thread doesn't become stuck should anything go wrong.